php/Blockonomics.php

```diff
@@ -547,6 +547,16 @@ public function fix_displaying_small_values($satoshi){
         }
     }
 
+    public function get_crypto_rate_from_params($value, $satoshi) {
+        // Crypto Rate is re-calculated here and may slightly differ from the rate provided by Blockonomics
+        // This is required to be recalculated as the rate is not stored anywhere in $order, only the converted satoshi amount is.
```

Comment:
The comment is misleading.
The main requirement of this function is to avoid scientific notification. Crypto rate can be calculated in JS also using satoshi/amount*1.0e8. However, this leads to sometimes scientific notation display like 2.0E107 .

TO avoid this server return a convert string can be be used for display both in JS / nojS templates

Reply:
This wasn't the motivation towards building this function, the same logic can be implemented in JS without showing the Scientific Notification by formatting number as we do here. The reason this method was required was:

1. We need to show the rate in both JS and No-JS that's why it makes sense to have it in the backend just like the payment_uri (https://github.com/blockonomics/woocommerce-plugin/pull/301/files/b20ff4224e5227ecf7de231838f56834b74897a9#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebR584-R586)

2. The primary concern for this method was the comment that was added and that was the difference in rates that Blockonomics Provide and the rate which is calculated here, and why it is being recalculated. to avoid confusion in the future.

But yes, we can add a comment mentioning why we added it in backend and not frontend directly,

![image](https://user-images.githubusercontent.com/22245433/209306216-62c30a90-b26d-40ad-9dcd-6faac879cac8.png)